app/src/main/java/org/mozilla/fenix/home/recentvisits/controller/RecentVisitsController.kt

```diff
@@ -16,66 +16,81 @@ import mozilla.components.feature.tabs.TabsUseCases.SelectOrAddUseCase
 import org.mozilla.fenix.library.history.toHistoryMetadata
 
 /**
- * An interface that handles the view manipulation of the history metadata in the Home screen.
+ * All possible updates following user interactions with the "Recent visits" section from the Home screen.
  */
-interface HistoryMetadataController {
+interface RecentVisitsController {
 
     /**
-     * @see [HistoryMetadataInteractor.onHistoryShowAllClicked]
+     * Callback for when the "Show all" link is clicked.
      */
     fun handleHistoryShowAllClicked()
 
     /**
-     * @see [HistoryMetadataInteractor.onRecentHistoryGroupClicked]
+     * Callback for when the user clicks on a specific [RecentHistoryGroup].
+     *
+     * @param recentHistoryGroup The just clicked [RecentHistoryGroup].
      */
     fun handleRecentHistoryGroupClicked(recentHistoryGroup: RecentHistoryGroup)
 
     /**
-     * @see [HistoryMetadataInteractor.onRemoveRecentHistoryGroup]
+     * Callback for when the user removes a certain [RecentHistoryGroup].
+     *
+     * @param groupTitle Title of the [RecentHistoryGroup] to remove.
      */
-    fun handleRemoveRecentHistoryGroup(searchTerm: String)
+    fun handleRemoveRecentHistoryGroup(groupTitle: String)
 
     /**
-     * @see [HistoryMetadataInteractor.onRecentHistoryHighlightClicked]
+     * Callback for when the user clicks on a specific [RecentHistoryHighlight].
+     *
+     * @param recentHistoryHighlight The just clicked [RecentHistoryHighlight].
      */
     fun handleRecentHistoryHighlightClicked(recentHistoryHighlight: RecentHistoryHighlight)
 
     /**
-     * @see [HistoryMetadataInteractor.onRemoveRecentHistoryHighlight]
+     * Callback for when the user removes a certain [RecentHistoryHighlight].
+     *
+     * @param highlightUrl Url of the [RecentHistoryHighlight] to remove.
      */
-    fun handleRemoveRecentHistoryHighlight(url: String)
+    fun handleRemoveRecentHistoryHighlight(highlightUrl: String)
 }
 
 /**
- * The default implementation of [HistoryMetadataController].
+ * The default implementation of [RecentVisitsController].
  */
-class DefaultHistoryMetadataController(
+class DefaultRecentVisitsController(
     private val store: BrowserStore,
     private val homeStore: HomeFragmentStore,
     private val selectOrAddTabUseCase: SelectOrAddUseCase,
     private val navController: NavController,
     private val storage: HistoryMetadataStorage,
     private val scope: CoroutineScope,
     private val metrics: MetricController
-) : HistoryMetadataController {
+) : RecentVisitsController {
 
+    /**
+     * Shows the history fragment.
+     */
     override fun handleHistoryShowAllClicked() {
         dismissSearchDialogIfDisplayed()
         navController.navigate(
             HomeFragmentDirections.actionGlobalHistoryFragment()
         )
     }
 
+    /**
+     * Navigates to the history metadata group fragment to display the group.
```

Comment:
Hm.. I was expecting we used this description in the controller interface above instead of having 2 different descriptions between the Interface and the Default implementation when I mentioned replacing the @see, but as is since the Interface still basically uses the description from the Interactor, there wasn't a point in replacing the @see since it is the same kDoc.

This isn't a blocker by any means.

Reply:
To keep the interface abstract it should not dictate behavior.
I see `Navigates to the history metadata group fragment` as an implementation detail.
If this description was set for interface's method declaration then all implementations should behave that way.